libcylinder/src/key.rs

```diff
@@ -77,6 +103,49 @@ impl PrivateKey {
     pub fn into_bytes(self) -> Vec<u8> {
         self.bytes
     }
+
+    #[cfg(feature = "pem")]
+    pub fn from_pem(s: &str) -> Result<Self, KeyParseError> {
+        let ec_key = EcKey::private_key_from_pem(s.as_bytes())?;
+
+        Ok(Self::new(ec_key.private_key().to_vec()))
+    }
+
+    #[cfg(feature = "pem")]
+    pub fn from_pem_with_password(s: &str, pw: &str) -> Result<Self, KeyParseError> {
+        let ec_key = EcKey::private_key_from_pem_passphrase(s.as_bytes(), pw.as_bytes())?;
+
+        Ok(Self::new(ec_key.private_key().to_vec()))
+    }
+
+    #[cfg(feature = "pem")]
+    fn to_ec_key(&self) -> Result<EcKey<EcPrivate>, KeyParseError> {
```

Comment:
We should drop this method, as it exposes the use of the openssl crate via the API.